callable/src/main/java/io/grpc/callable/Callable.java

```diff
@@ -0,0 +1,423 @@
+   * which is bound to a channel.
+   */
+  public ResponseT call(RequestT request) {
+    return call(null, request);
+  }
+
+  /**
+   * Convenience method to run a unary callable asynchronously. If no channel is provided,
+   * the callable must be bound to one.
+   */
+  public void asyncCall(@Nullable Channel channel, RequestT request,
+      StreamObserver<ResponseT> responseObserver) {
+    ClientCalls.asyncUnaryCall(newCall(channel), request, responseObserver);
+  }
+
+  /**
+   * Convenience method to run a unary callable asynchronously, without channel. Requires a callable
+   * which is bound to a channel.
+   */
+  public void asyncCall(RequestT request, StreamObserver<ResponseT> responseObserver) {
+    asyncCall(null, request, responseObserver);
+  }
+
+  /**
+   * Convenience method to run a unary callable returning a future. If no channel is provided,
+   * the callable must be bound to one.
+   */
+  public ListenableFuture<ResponseT> futureCall(@Nullable Channel channel, RequestT request) {
+    return ClientCalls.futureUnaryCall(newCall(channel), request);
+  }
+
+  /**
+   * Convenience method to run a unary callable returning a future, without a channel. Requires a
+   * callable which is bound to a channel.
+   */
+  public ListenableFuture<ResponseT> futureCall(RequestT request) {
+    return futureCall(null, request);
+  }
+
+  /**
+   * Convenience method for a blocking server streaming call. If no channel is provided,
+   * the callable must be bound to one.
+   *
+   * <p>Returns an iterable for the responses. Note the returned iterable can be used only once.
+   * Returning an Iterator would be more precise, but iterators cannot be used in Java for loops.
+   */
+  public Iterable<ResponseT> iterableResponseStreamCall(@Nullable Channel channel,
+      RequestT request) {
+    final Iterator<ResponseT> result =
+        ClientCalls.blockingServerStreamingCall(newCall(channel), request);
+    return new Iterable<ResponseT>() {
+      @Override
+      public Iterator<ResponseT> iterator() {
+        return result;
+      }
+    };
+  }
+
+  /**
+   * Convenience method for a blocking server streaming call, without a channel. Requires a
+   * callable which is bound to a channel.
+   *
+   * <p>Returns an iterable for the responses. Note the returned iterable can be used only once.
+   * Returning an Iterator would be more precise, but iterators cannot be used in Java for loops.
+   */
+  public Iterable<ResponseT> iterableResponseStreamCall(RequestT request) {
+    return iterableResponseStreamCall(null, request);
+  }
+
+  // Creation
+  // ========
+
+  /**
+   * Returns a callable which executes the described method.
+   *
+   * <pre>
+   *  Response response = Callable.create(SerivceGrpc.CONFIG.myMethod).call(channel, request);
+   * </pre>
+   */
+  public static <RequestT, ResponseT> Callable<RequestT, ResponseT>
+      create(MethodDescriptor<RequestT, ResponseT> descriptor) {
```

Comment:
add variant with CallOptions

should there be a '.with' pattern to permute the CallOptions easily